cdxev/set.py

```diff
@@ -28,6 +31,105 @@ class Context:
     sbom: dict
 
 
+@dataclass(init=True, frozen=True)
+class CoordinatesWithVersionRange(Coordinates):
+    """
+    This class inherits from cdxev.auxiliary.identity.Coordinates
+    and extends it to be able to handle version ranges.
+    It achieves this by introducing the attribute 'version_range'
+    of type univers.version_range.VersionRange.
+
+    This class's comparator is compatible with Coordinates objects.
+    If the 'name' and 'group' match,
+    it will check if the provided version of the coordinates object
+    is contained in this class instance's 'version_range'
+    and return True or False depending on the result."
+    """
+
+    version_range: univers.version_range.VersionRange
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, CoordinatesWithVersionRange):
+            for field_ in fields(self):
+                if getattr(self, field_.name) != getattr(other, field_.name):
+                    return False
+            return True
+
+        if isinstance(other, Coordinates):
+            for field_ in fields(other):
+                if field_.name != "version" and getattr(self, field_.name) != getattr(
+                    other, field_.name
+                ):
+                    return False
+
+            if other.version is not None:
+                if (
+                    self.version_range.version_class(other.version)
```

Comment:
This is a minor user-experience problem. Assume that the SBOM contains inconsistent data or the user makes a mistake in specifying the scheme of their version range.
This line creates a version object for each component which has the same `name` and `group` as the component we want to update. If the component's version specified in the SBOM cannot be parsed into the versioning scheme the user gave in their range, the tool will crash with an ugly error. 
You should catch that case and either raise an `AppError` (terminates the program with a nice error message) or simply output a warning.

Reply:
I chose to interpret it as "no match" and put out a warning containing the provided version type and the ones the given version is valid under.

Reply:
But it might be a bit overboard.